01.multithreading/MultiThreading.Task4.Threads.Join/Program.cs

```diff
@@ -0,0 +1,62 @@
+﻿/*
+ * 4.	Write a program which recursively creates 10 threads.
+ * Each thread should be with the same body and receive a state with integer number, decrement it,
+ * print and pass as a state into the newly created thread.
+ * Use Thread class for this task and Join for waiting threads.
+ * 
+ * Implement all of the following options:
+ * - a) Use Thread class for this task and Join for waiting threads.
+ * - b) ThreadPool class for this task and Semaphore for waiting threads.
+ */
+
+using System;
+using System.Threading;
+
+namespace MultiThreading.Task4.Threads.Join
+{
+    class Program
+    {
+        private const int StateNumber = 10;
+        private static readonly SemaphoreSlim _semaphore = new(3);
+
+        static void Main(string[] args)
+        {
+            Console.WriteLine("4.	Write a program which recursively creates 10 threads.");
+            Console.WriteLine("Each thread should be with the same body and receive a state with integer number, decrement it, print and pass as a state into the newly created thread.");
+            Console.WriteLine("Implement all of the following options:");
+            Console.WriteLine();
+            Console.WriteLine("- a) Use Thread class for this task and Join for waiting threads.");
+            Console.WriteLine("- b) ThreadPool class for this task and _semaphore for waiting threads.");
+
+            Console.WriteLine();
+
+            ProcThread(StateNumber);
+            ProcThreadPool(StateNumber);
+        }
+
+        private static void ProcThread(object obj)
+        {
+            var number = (int)obj;
+            Console.WriteLine("Threads - " + number);
+            if (number > 0)
```

Comment:
Take a look how many thread did you create) Seems 11, the right condition is number > 1.